### PR TITLE
Closes #3168 Add the server protocol to our analytics

### LINKED ISSUE #3168: Add HTTP protocol version to Rocket Analytics
**Describe the solution you'd like**
To help us to make some important decisions related to combine options, we will need to know the HTTP protocol version used by ours users.

Having the version like 1.1, 2 and 3 will be enough.

### PR DESCRIPTION
**Pull request by Tabrisrp**
Closes #3168 

### Code changes
Send the server protocol to our analytics. We parse the `SERVER_PROTOCOL` value to only keep the version number based on the `http\/(?.*)$` pattern.

### Acceptance criteria
The server protocol is correctly sent to Mixpanel, using the `http_protocol` index, and the version number as value

### CODE REVIEW

File: inc/admin/admin.php [hunk 1]
```diff
@@ -397,10 +397,11 @@ function rocket_analytics_data() {
...
 
-	$theme              = wp_get_theme();
-	$data               = array_diff_key( get_option( WP_ROCKET_SLUG ), $untracked_wp_rocket_options );
-	$locale             = explode( '_', get_locale() );
-	$data['web_server'] = 'Unknown';
+	$theme                 = wp_get_theme();
+	$data                  = array_diff_key( get_option( WP_ROCKET_SLUG ), $untracked_wp_rocket_options );
+	$locale                = explode( '_', get_locale() );
+	$data['web_server']    = 'Unknown';
+	$data['http_protocol'] = '';
```
Comment (engahmeds3ed) [inline inc/admin/admin.php:404]: I think no need to use this initialization as always the function `wp_get_server_protocol` will return valid value.

**Review by engahmeds3ed — APPROVED** [review]
Only one optimization

**Comment by Tabrisrp** [comment]
@wp-media/qa ready for QA

**Comment by piotrbak** [comment]
@Tabrisrp Could you give me access to the Mixpanel?

**Comment by Tabrisrp** [comment]
You should have it in Bitwarden

**Comment by piotrbak** [comment]
@Tabrisrp I'm wondering how accurate data we're going to receive. 

We'll get the HTTP/1 protocol for all sites using Cloudflare:
image
Probably because all connections between Cloudflare and the origin server are done via HTTP/1:
https://support.cloudflare.com/hc/en-us/articles/200168076-Are-the-HTTP-2-or-SPDY-protocols-supported-between-Cloudflare-and-the-origin-server-#6ncFUWOVRaVtPzYN1euBIC

We'll get the same results for the servers when there's Nginx as a reverse proxy for Apache working.

I also tested it also on the HTTP3:
image

It's not even possible to get an HTTP/3 protocol from this function, so it'll become a default HTTP/1.0:
https://developer.wordpress.org/reference/functions/wp_get_server_protocol/

When it comes to the sending data itself. I can see that the protocol data is present in the sent array, but I'm not able to find it in the Mixpanel. It's not present in the list:
image
Do you know if I need to set something up there?

**Comment by Tabrisrp** [comment]
@GeekPress From QA, accuracy of the data for this might not be good, do we still want to collect it?

**Comment by GeekPress** [comment]
@Tabrisrp Indeed if the accuracy is bad, we don't have the interest to collect it. Let's forget that for now?

ping @arunbasillal :)

**Comment by arunbasillal** [comment]
Really amazing QA there @piotrbak 

I agree with @GeekPress Bad accuracy can lead to bad data and bad judgements. Better not to have it. A lot of users use Cloudflare, so it's not negligible.

**Pull request closed without merging**